code/game/machinery/buttons.dm

```diff
@@ -89,12 +89,32 @@
 		if(!device && !board && W.tool_behaviour == TOOL_WRENCH)
 			to_chat(user, "<span class='notice'>You start unsecuring the button frame...</span>")
 			W.play_tool_sound(src)
-			if(W.use_tool(src, user, 40))
+			if(W.use_tool(src, user, 4 SECONDS))
 				to_chat(user, "<span class='notice'>You unsecure the button frame.</span>")
 				transfer_fingerprints_to(new /obj/item/wallframe/button(get_turf(src)))
 				playsound(loc, 'sound/items/deconstruct.ogg', 50, 1)
 				qdel(src)
 
+		if(W.tool_behaviour == TOOL_MULTITOOL)
+			if(istype(device, /obj/item/assembly/control)) // User Feedback
+				var/obj/item/multitool/P = W
+				if(!id) // Generate New ID if none exists
+					id = rand(1, 25565) // rare enough that ids should never conflict
+					to_chat(user, "<span class='notice'>No ID found. Generating New ID</span>")
+
+				P.buffer = id
+				to_chat(user, "<span class='notice'>You link the button to the [P].</span>")
+				setup_device() // Has to be done. It sets the signaller up
+			else
+				to_chat(user, "[device ? "" : "<span class='notice'>No blast door controller found</span>"]")
```

Comment:
why not just an if statement?

Reply:
er what. it just tells you if you dont have a blast door controller in it

Reply:
yeah but then why do you have the device ? statement, either remove that or add an if statement before the to_chat
(preferably the first option)